src/languages/typescript.js

```diff
@@ -35,9 +35,9 @@ export default function(hljs) {
   ];
   var KEYWORDS = {
     $pattern: ECMAScript.IDENT_RE,
-    keyword: ECMAScript.KEYWORDS.concat(TS_SPECIFIC_KEYWORDS).join(" "),
+    keyword: ECMAScript.KEYWORDS.concat(TYPES).concat(TS_SPECIFIC_KEYWORDS).join(" "),
     literal: ECMAScript.LITERALS.join(" "),
-    built_in: ECMAScript.BUILT_INS.concat(TYPES).join(" ")
+    built_in: ECMAScript.BUILT_INS.join(" ")
```

Comment:
This was intentional - to classify the types differently than keywords.  This is in line with how all the built-in JS classes are classified (which are also technically types): as built-ins.

Reply:
Yeah, my misunderstanding of the TS context, removed it because it's indeed nonsense